authentication/views.py

```diff
@@ -137,41 +169,51 @@ def send(self, to, *args, **kwargs):
-        when redux-hammock is changed to support 204's.
-        """
+    1. Returns 404 if the EMAIL_AUTH feature is not enabled
+    2. In version 0.30.0, the fetch function in redux-hammock does not handle responses
+    with empty response data. Djoser returns 204's with empty response data, so we are
+    coercing that to a 200 with an empty dict as the response data. This can be removed
+    when redux-hammock is changed to support 204's.
+    """
+    def _action(self, serializer):  # pylint: disable=missing-docstring
         if not features.is_enabled(features.EMAIL_AUTH):
             return Response(status=status.HTTP_404_NOT_FOUND)
-        response = super().post(request)
+        response = super()._action(serializer)
         if response.status_code == status.HTTP_204_NO_CONTENT:
             return Response({}, status=status.HTTP_200_OK)
         return response
 
 
-class CustomPasswordResetConfirmView(DjoserPasswordResetConfirmView):
+class CustomPasswordResetView(CustomDjoserAPIView, DjoserPasswordResetView):
+    """Custom view to modify base functionality in Djoser's PasswordResetView class"""
+    pass
+
+
+class CustomPasswordResetConfirmView(CustomDjoserAPIView, DjoserPasswordResetConfirmView):
     """Custom view to modify base functionality in Djoser's PasswordResetConfirmView class"""
-    def post(self, request):
-        """
-        Overrides djoser.views.PasswordResetView#post and adds two bits of logic:
+    pass
+
 
-        1. Returns 404 if the EMAIL_AUTH feature is not enabled
-        2. In version 0.30.0, the fetch function in redux-hammock does not handle responses
-        with empty response data. Djoser returns 204's with empty response data, so we are
-        coercing that to a 200 with an empty dict as the response data. This can be removed
-        when redux-hammock is changed to support 204's.
+class CustomSetPasswordView(CustomDjoserAPIView, DjoserSetPasswordView):
+    """Custom view to modify base functionality in Djoser's SetPasswordView class"""
+    permission_classes = (IsAuthenticated,)
+
+    def _action(self, serializer):
         """
-        if not features.is_enabled(features.EMAIL_AUTH):
-            return Response(status=status.HTTP_404_NOT_FOUND)
-        response = super().post(request)
-        if response.status_code == status.HTTP_204_NO_CONTENT:
-            return Response({}, status=status.HTTP_200_OK)
+        Overrides CustomDjoserAPIView._action. If the settings indicate that the user
+        should not be logged out after a successful password change, the session is
+        updated. Without this explicit refresh, the user's session would be invalid
+        and they would be logged out.
+        """
+        response = super()._action(serializer)
+        if (
+                response.status_code in (status.HTTP_200_OK, status.HTTP_204_NO_CONTENT) and
+                settings.DJOSER['LOGOUT_ON_PASSWORD_CHANGE'] is False
```

Comment:
Would this setting ever be `True`?  If not, maybe remove the check for it here?

Reply:
And remove the setting altogether?

Reply:
I included this just to guard against a future case where we change this setting, but looking at it now I agree with you. I'll remove this logic. I'll keep the setting just to be explicit about the behavior we want